source/codegen/metadata_mutation.py

```diff
@@ -59,6 +59,85 @@ def mark_mapped_enum_params(parameters, enums):
             del param['enum']
             param['mapped-enum'] = enum_name
 
+
+def get_grpc_type_from_ivi(type, is_array, driver_name_pascal):
+    add_repeated = is_array
+    if 'ViSession' in type:
+        type = 'nidevice_grpc.Session'
+    if 'ViBoolean' in type:
+        type = 'bool'
+    if 'ViReal64' in type:
+        type = 'double'
+    if 'ViInt32' in type:
+        type = 'sint32'
+    if 'ViConstString' in type:
+        type = 'string'
+    if 'ViString' in type:
+        type = 'string'
+    if 'ViRsrc' in type:
+        type = 'string'
+    if 'ViChar' in type:
+        if is_array == True:
+            add_repeated = False
+            type = 'string'
+        else:
+            type = 'uint32'
+    if 'ViReal32' in type:
+        type = 'float'
+    if 'ViAttr' in type:
+        type = driver_name_pascal + "Attributes"
+    if 'ViInt8' in type:
+        if is_array == True:
+            type = "bytes"
+            add_repeated = False
+        else:
+            type = 'uint32'
+    if 'void*' in type:
+        type = 'fixed64'
+    if 'ViInt16' in type:
+        type = 'sint32'
+    if 'ViInt64' in type:
+        type = 'int64'
+    if 'ViUInt16' in type:
+        type = 'uint32'
+    if 'ViUInt32' in type:
+        type = 'uint32'
+    if 'ViUInt64' in type:
+        type = 'uint64'
+    if 'ViUInt8' in type:
+        if is_array == True:
+            type = "bytes"
+            add_repeated = False
+        else:
+            type = 'uint32'
+    if 'ViStatus' in type:
+        type = 'sint32'
+    if 'ViAddr' in type:
+        type = 'fixed64'
+    if 'int' == type:
+        type = 'sint32'
+    if "[]" in type:
+        type = type.replace("[]", "")
+
+    return "repeated " + type if add_repeated else type
+
+
+def populate_grpc_types(parameters, config):
```

Comment:
Note: this method moves the pre-existing `get_parameter_type` to metadata mutation (setting the result to grpc_type) and augments it with support for a `type_to_grpc_type`.

Right now `type_to_grpc_type` is all-or-nothing (or really all-or-ivi). We can change that as it develops (or even use something more sophisticated to calculate mappings). But, for now, that makes it a little easier to catch "oops I forgot to add a new type to DAQ".